lib/dfe_sign_in/api/user.rb

```diff
@@ -58,20 +58,32 @@ def has_role?(role_code)
       end
 
       def role_codes
+        unless service_access?
+          raise ExternalServerError, "#{response.code}: #{response.body}" unless response.code.eql?("200")
+        end
+
         body["roles"].map { |r| r["code"] }
       end
 
+      def service_access?
+        response.code == "200"
+      end
+
+      def service_error?
+        response.code == "500"
+      end
```

Comment:
don't think you're actually using this any where?

Reply:
No, I guess YAGNI applies here